File: 백준/Silver/2606. 바이러스/바이러스.py
N = int(input()) # 컴퓨터의 수
M = int(input()) # 네트워크 상에 직접 연결되어 있는 컴퓨터 쌍의 수

graph = [[] for _ in range(N+1)]	



for n in range(M):
    a, b = map(int, input().split())
    # 양방향
    graph[a].append(b)
    graph[b].append(a)

visited = [False] * (N+1)
count = -1


def dfs(v):
    visited[v] = True
    global count
    count += 1
    for i in graph[v]:
        if not visited[i]:
            dfs(i)

dfs(1)
print(count)

File: 백준/Silver/2606. 바이러스/바이러스.java
import java.io.BufferedReader;
import java.io.InputStreamReader;
import java.util.*;

public class Main {
    static ArrayList<Integer>[] graph;
    static boolean[] visited;
    static int computer;

    public static void main(String[] args) throws Exception {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st;

        int c = Integer.parseInt(br.readLine()); // computer
        int n = Integer.parseInt(br.readLine()); // line

        graph = new ArrayList[c+1];
        visited = new boolean[c+1];


        for (int i = 1; i <= c; i++) {
            graph[i] = new ArrayList<>();
        }

        // input
        for (int i = 0; i < n; i++) {
            st = new StringTokenizer(br.readLine(), " ");
            int x = Integer.parseInt(st.nextToken());
            int y = Integer.parseInt(st.nextToken());

            // 양방향
            graph[x].add(y);
            graph[y].add(x);
        }

//        System.out.println(Arrays.toString(graph));

        dfs(1);

        System.out.println(computer-1);

    }

    static void dfs(int a){
        visited[a] = true; // 방문 처리
        computer++;

        for(int next: graph[a]){ // 하나씩 전부 방문
            if(!visited[next]){
                dfs(next);
            }
        }
    }

}
